python/MooseDocs/extensions/bibtex.py

```diff
@@ -227,8 +227,11 @@ def createHTML(self, parent, token, page):
             author = LatexNodes2Text().latex_to_text(author)
 
             form = '{}, {}' if citep else '{} ({})'
-            html.Tag(parent, 'a', href='#{}'.format(key),
-                     string=form.format(author, entry.fields['year']))
+            year = entry.fields.get('year', None)
+            if year is None:
+                LOG.warning("Unable to locate year for bibtex entry '{}'".format(entry.key))
+                year = '????'
```

Comment:
Would it be better to just leave year empty? I.e. `year = ''`. Or is this the behavior that someone has asked for?

Reply:
Nevermind. I see the issue with inline render.

Reply:
I think I will change this to an error, but I need to go fix an entry. You can hold off on merging.